wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -131,12 +141,15 @@ func (v *Wrapper) SetConfig(ctx context.Context, opt ...wrapping.Option) (*wrapp
 	if azResource == "" {
 		azResource = opts.withResource
 		if azResource == "" {
-			azResource = v.environment.KeyVaultDNSSuffix
+			_, azResource, err = cloudConfigAndKeyVaultDNSSuffixFromName(envName)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	v.environment.KeyVaultDNSSuffix = azResource
+	v.keyVaultDNSSuffix = azResource
 	v.resource = "https://" + azResource + "/"
-	v.environment.KeyVaultEndpoint = v.resource
+	v.keyVaultEndpoint = v.resource
```

Comment:
`keyVaultEndpoint` looks unused after it's set here? I wonder if we're supposed to set the endpoint on the `cloudConfig` similar to the environment before.